scripts/bulk_load.py

```diff
@@ -0,0 +1,111 @@
+"""Runs database write and read tests with many tables.
+
+Findings:
+- DuckDB is significantly faster than SQLite for both data ingestion and reads.
+- SQLite benefits significantly from indexes.
+
+Usage: python scripts/perf_many_tables.py
+
+Results on 12/27/2024:
+Run test with engine_url='duckdb:///:memory:' count=1000 length=8760 create_index=True
+duration_write=1.0231969356536865
+duration_read=5.646288871765137 avg=0.005646288871765137
+Run test with engine_url='sqlite:///:memory:' count=1000 length=8760 create_index=True
+duration_write=46.84509587287903
+duration_read=14.509371757507324 avg=0.014509371757507324
+Run test with engine_url='duckdb:///:memory:' count=1000 length=8760 create_index=False
+duration_write=1.1198809146881104
+duration_read=0.4036238193511963 avg=0.0004036238193511963
+Run test with engine_url='sqlite:///:memory:' count=1000 length=8760 create_index=False
+duration_write=46.14994025230408
+duration_read=199.04997396469116 avg=0.19904997396469115
+"""
+
+import time
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Generator
+
+import numpy as np
+import pandas as pd
+from sqlalchemy import create_engine, select
+
+from chronify.models import TableSchema
+from chronify.store import Store
+from chronify.time_configs import DatetimeRange
+
+
+GENERATOR_TIME_SERIES_FILE = "tests/data/gen.csv"
+
+
+def make_tables(
+    resolution: timedelta, length: int, count: int
+) -> Generator[pd.DataFrame, None, None]:
+    start = datetime(2020, 1, 1)
+    end = start + resolution * (length - 1)
+    timestamps = pd.date_range(start, end, freq=resolution)
```

Comment:
NIT, you could have just done `pd.date_range(start, periods=length, freq=resolution)`.